plugins/inventory/stackpath_compute.py

```diff
@@ -0,0 +1,242 @@
+
+import json
+
+from ansible.errors import AnsibleError
+from ansible.module_utils._text import to_native, to_text
+from ansible.plugins.inventory import (
+    BaseInventoryPlugin,
+    Constructable,
+    Cacheable
+)
+from ansible.utils.display import Display
+
+try:
+    import requests
+except ImportError:
+    raise AnsibleError(
+        'The stackpath_compute dynamic inventory plugin requires requests.'
+    )
+
+DOCUMENTATION = '''
+    name: stackpath_compute
+    plugin_type: inventory
+    short_description: StackPath Edge Computing inventory source
+    requirements:
+        - requests
+    extends_documentation_fragment:
+        - inventory_cache
+        - constructed
+    description:
+        - Get inventory hosts from StackPath Edge Computing.
+        - Uses a YAML configuration file that ends with stackpath_compute.(yml|yaml).
+    options:
+        plugin:
+            description: >
+                A token that ensures this is a source file for the 'stackpath_compute' plugin.
+            required: True
+            choices: ['stackpath_compute']
+        client_id:
+            description: >
+                An OAuth client ID generated from the API Management section of the StackPath customer portal
+                U(https://control.stackpath.net/api-management)
+            requierd: True
+        client_secret:
+            description: >
+                An OAuth client secret generated from the API Management section of the StackPath customer portal
+                U(https://control.stackpath.net/api-management)
+            required: True
+        stack_ids:
+            description: >
+                A list of Stack IDs to query instnaces in. If no entry then get instances in all stacks on the account
+                U(https://developer.stackpath.com/docs/en/getting-started/#get-your-stack-id)
+            required: False
+        use_internal_ip:
+            description: >
+                Whether or not to use internal IP addresses, yes to use internal addresses, no to use external addresses.
+                Defaults to external
+            requiered: False
+            choices: ['yes', 'no']
+'''
+
+EXAMPLES = '''
+
+# example using credentials to fetch all workload instances in a stack.
+---
+plugin: stackpath_compute
+client_id: my_client_id
+client_secret: my_client_secret
+stack_ids:
+- my_first_stack_id
+- my_other_stack_id
+use_internal_ip: no
+
+'''
+
+display = Display()
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+
+    NAME = 'stackpath_compute'
+
+    def __init__(self):
+        super(InventoryModule, self).__init__()
+
+        # credentials
+        self.client_id = None
+        self.client_secret = None
+        self.stack_id = None
+        self.api_host = "https://gateway.stackpath.com"
+        self.group_keys = [
+            "stackId", "workloadId", "cityCode",
+            "countryCode", "continent", "target", "name", "workloadSlug"]
```

Comment:
```suggestion
        self.group_keys = [
            "stackId",
            "workloadId",
            "cityCode",
            "countryCode",
            "continent",
            "target",
            "name",
            "workloadSlug",
        ]
```
just considered as a good practice